### PR TITLE
Enable caching using memcached

### PR DESCRIPTION
**Pull request by kajinamit**
This introduces the capability to use memcached as cache backend, to reproduce the setting in TripleO.

Enabling caching in keystone provides large performance benefit especially in token validation.

### CODE REVIEW

**Comment by kajinamit** [comment]
/test keystone-operator-build-deploy-kuttl

failed to wait for the created cluster claim to become ready: timed out waiting for the condition

File: api/v1beta1/conditions.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ const (
 
 	// KeystoneServiceOSUserReadyCondition Status=True condition which indicates if the service user got created in the keystone instance is ready/was successful
 	KeystoneServiceOSUserReadyCondition condition.Type = "KeystoneServiceOSUserReady"
+
+	// KeystoneMemcachedReadyCondition - Indicates the Keystone memcached service is ready to be consumed by Keystone
+	KeystoneMemcachedReadyCondition condition.Type = "KeystoneMemcached"
```
Comment (abays) [inline api/v1beta1/conditions.go:40]: Perhaps the value should `KeystoneMemcachedReady` to keep with the `<ServiceName><Component>Ready` pattern?

Comment (kajinamit) [reply api/v1beta1/conditions.go]: Yes, that makes sense. I'm fixing it in horizon-operator as well. https://github.com/openstack-k8s-operators/horizon-operator/pull/157

**Comment by kajinamit** [comment]
We probably should merge https://github.com/openstack-k8s-operators/keystone-operator/pull/268 first then I'll update the test case to include this change.

**Comment by kajinamit** [comment]
/test keystone-operator-build-deploy-kuttl

wired. looks like we are again facing problems with storage cleanup ?

**Comment by abays** [comment]
> /test keystone-operator-build-deploy-kuttl
> 
> wired. looks like we are again facing problems with storage cleanup ?

https://github.com/openstack-k8s-operators/install_yamls/pull/368 is now merged.  Going to retest to see if it helps.

/test keystone-operator-build-deploy-kuttl

**Comment by kajinamit** [comment]
/test horizon-operator-build-deploy-kuttl

https://github.com/openstack-k8s-operators/install_yamls/pull/374 was merged

**Comment by kajinamit** [comment]
/test keystone-operator-build-deploy-kuttl

oops wrong copy-paste.

File: controllers/keystoneapi_controller.go [hunk 2]
```diff
@@ -96,6 +98,7 @@ type KeystoneAPIReconciler struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete;
 // +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;list;watch;create;update;patch;delete;
 // +kubebuilder:rbac:groups=mariadb.openstack.org,resources=mariadbdatabases,verbs=get;list;watch;create;update;patch;delete;
+// +kubebuilder:rbac:groups=memcached.openstack.org,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
```
Comment (abays) [inline controllers/keystoneapi_controller.go:101]: Does the operator need `create;update;patch;delete` here?  Same with `list`.  Seems like it only `get`s and `watch`es the `Memcached` resource.

Comment (kajinamit) [reply controllers/keystoneapi_controller.go:101]: This operator does not create or update Memcached resource. It requires that the user creates it seprately (eg via top-level control plane resource) and give the reference by memcachedInstance.

Comment (abays) [reply controllers/keystoneapi_controller.go:101]: Right.  That's why I'm asking if we can remove the `create;update;patch;delete` and `list` from the annotated permissions.  It seems we don't need them here.

Comment (kajinamit) [reply controllers/keystoneapi_controller.go]: Sorry I thought I updated this but it was wrong. Fixed it in the latest version.

Comment (kajinamit) [reply controllers/keystoneapi_controller.go]: fyi: I'm fixing this in horizon-operator by https://github.com/openstack-k8s-operators/horizon-operator/pull/168

**Pull request merged**